### PR TITLE
feat: Add logging for debugging

### PR DESCRIPTION
**Pull request by rmi22186**
## Summary
This PR provides additional logging to allow customers to confirm what method and what arguments are being passed to each method.

As an example, here is what logging a purchase looks like in the logs:
image

## Testing Plan
Tested every logging method locally

## Reference Issue
- Closes https://go.mparticle.com/work/SQDSDKS-4443

### CODE REVIEW

File: src/BrazeKit-dev.js [hunk 1]
```diff
@@ -141,6 +154,13 @@ var constructor = function() {
                     ", removeUserAttribute or setUserAttribute must set 'dob' to a date"
                 );
             } else {
+                kitLogger(
+                    'appoy.getUser().setDateOfBirth',
+                    value.getFullYear(),
+                    value.getMonth() + 1,
+                    value.getDate()
+                );
+                s;
```
Comment (alexs-mparticle) [inline src/BrazeKit-dev.js:163]: typo?

Comment (rmi22186) [reply src/BrazeKit-dev.js:163]: good catch. thanks

**Comment by rmi22186** [comment]
WAIT TO MERGE - pending Core SDK to set logger on kits.

File: src/BrazeKit-dev.js [hunk 2]
```diff
@@ -512,6 +592,24 @@ var constructor = function() {
     this.onUserIdentified = onUserIdentified;
     this.removeUserAttribute = removeUserAttribute;
     this.decodeClusterSetting = decodeClusterSetting;
+
+    function kitLogger(method) {
```
Comment (alexs-mparticle) [inline src/BrazeKit-dev.js:596]: Can you provide a comment of the expected output of this function so future devs can use it as a cheat sheet?

Comment (rmi22186) [reply src/BrazeKit-dev.js:596]: good call, will do

Comment (alexs-mparticle) [inline src/BrazeKit-dev.js:596]: Feels like a nitpick, but I would also wrap a small tests around this logger, in case we break it in the future.

Comment (rmi22186) [reply src/BrazeKit-dev.js:610]: updated.

**Review by alexs-mparticle — APPROVED** [review]

**Pull request merged**